pytest-embedded/pytest_embedded/unity.py

```diff
@@ -324,5 +326,11 @@ def merge(self, junit_files: List[str]):
             if int(junit_parent.attrib['failures']) > 0:
                 self.failed = True
 
+        for testcase in self.junit.findall('.//testcase'):
+            # Add the 'pytest_case_name' attribute for each test case to enable later extraction
+            # of the app_path from the pytest_case_name to app_path mapping
+            if 'pytest_case_name' not in testcase.attrib:
+                testcase.attrib['pytest_case_name'] = testcase.attrib.get('name')
```

Comment:
question, why can't we use the `attrib['name']` here?

Reply:
The issue is that in some cases, the pytest test case name is being replaced by the C test case name, resulting in the name being absent in the mapping.

Reply:
Unfortunately, this suggestion isn't viable. For example,` self.testsuite.add_unity_test_cases` is executed only within the `expect_unity_test_output` context. However, there are numerous scenarios where users call `expect_exact`. Additionally, I tested it on the provided tests in this PR, and in both cases, the proposed solution either fails to work or requires significantly more changes compared to the current implementation.

Reply:
I made this work in #338, please take a look. I'll close this PR.